integration-tests/pkg/types/network.go

```diff
@@ -1,103 +1,132 @@
 package types
 
 import (
-	"fmt"
+	"bytes"
+	"net"
 	"sort"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 
 	sensorAPI "github.com/stackrox/rox/generated/internalapi/sensor"
-	utils "github.com/stackrox/rox/pkg/net"
 )
 
 const (
 	NilTimestamp = "<nil>"
 )
 
-type NetworkInfo struct {
-	LocalAddress   string
-	RemoteAddress  string
-	Role           string
-	SocketFamily   string
-	CloseTimestamp string
+var (
+	NotNilTimestamp = timestamppb.New(time.Date(2025, 1, 1, 0, 0, 0, 0, time.UTC))
+)
+
+type NetworkConnectionBatch []*sensorAPI.NetworkConnection
+
+func IsActive(conn *sensorAPI.NetworkConnection) bool {
+	// no close timestamp means the connection is open, and active
+	return conn.GetCloseTimestamp() == nil
 }
 
-type NetworkInfoBatch []NetworkInfo
+// Equal is not called directly because it returns false when close timestamps have different non-nil values.
+func EqualNetworkConnection(conn1 *sensorAPI.NetworkConnection, conn2 *sensorAPI.NetworkConnection) bool {
+	copyConn1 := conn1.CloneVT()
+	copyConn2 := conn2.CloneVT()
```

Comment:
I don't think we need to always copy, and in fact we're copying up to four times in this function (if either copy has no close timestamp we copy it again)

We can drop both of these CloneVT calls, and just pass the conn1 and conn2 pointers to `adjustNetworkConnectionForComparison` which will do the clone only if we need to modify

Reply:
I am no longer using `CloneVT` when it is not needed.